### ISSUE TITLE
KFAC converges fast but validation accuracy is not so good

### ISSUE THREAD

**Issue by jinxin0924** [tensorflow/kfac#12]
Hi,
I have trained Resnet-20 on cifar-10 using KFAC several times and all the experiments reveals that although KFAC converges faster, but the final validation accuracy is not as good as momentum-optimizer's. 
Below is the optimization performance of KFAC and Momentum training Resnet-20 on cifar10:
![kfac_momentum](https://user-images.githubusercontent.com/8096033/40903710-e91fb670-680a-11e8-8a32-92a75cfd5ae2.PNG)
Is it common that KFAC converges faster but get worse performance? Can I tune some parameters to get better performance? Now I only tried different learning rates but kept cov_ema_decay and damping unchanged, as same as in [convnet.py](https://github.com/tensorflow/kfac/blob/master/kfac/examples/convnet.py). Could you give some suggestions to tune these two parameters? Thanks!

**Comment by jinxin0924** [comment]
@james-martens , could you give me some suggestions?

**Comment by james-martens** [comment]
Hi Jin Xin,

It's possible that for some architectures KFAC might overfit slightly more, especially if you used larger mini-batches.  Sometimes running SGD for the first epoch can eliminate this effect.

As far tuning the parameters, learning rate and damping are probably the most important.  Momentum can often be left at its default value of 0.9.  Generally the higher you make the learning rate the higher the damping must be.  But their relationship is complicated.  Lower damping generally leads to faster optimization but if it's too low this can cause instability and divergence.

One option you might consider is the automatic damping tuning feature that is implemented in the case.  You might also want to use the "norm clipping" feature built into the code which might let you get away with a larger learning rate / lower damping value.

I hope that helps.

James

**Comment by jinxin0924** [comment]
@james-martens Thanks for your suggestions! I will try them.

**Comment by duckworthd** [comment]
Closing as there's no further actions at the time.

**Comment by se4u** [comment]
@jinxin0924 I was wondering if you tried james' suggestions and what you learned from your experiments, about how KFAC's risk can be kept low? 

Did any of the regularization methods such as dropout, gradient clipping, weight decay, smaller mini-batch, or optimization tricks such as  periodically reverting to SGD for an epoch, learning rate annealing, or high damping (which decreases the influence of the hessian approximation in computing the update from the gradient) work for you?

**Comment by jinxin0924** [comment]
@se4u I tried to tune damping, learning rate and weight decay, but all the experiments with KFAC performed not as good as momentum's. I guess it's because of the architecture, as I only tried Resnet-20 on cifar10. Hope this may help you!

**Comment by kudkudak** [comment]
It is interesting that weight decay doesn't help given https://openreview.net/forum?id=B1lz-3Rct7?. Perhaps there are some details in the way they apply WD which differ from your setting?
